typescript/web/src/components/labelling-tool/view-tool-bar/full-screen-tool.tsx

```diff
@@ -0,0 +1,127 @@
+import { RefObject, useState } from "react";
+import {
+  IconButton,
+  chakra,
+  Tooltip,
+  useColorModeValue as mode,
+} from "@chakra-ui/react";
+import { RiFullscreenLine, RiFullscreenExitLine } from "react-icons/ri";
+
+import { isInWindowScope } from "../../../utils/detect-scope";
+
+const FullscreenIcon = chakra(RiFullscreenLine);
+const FullscreenExitIcon = chakra(RiFullscreenExitLine);
+
+// For Full screen related stuff,
+// See https://github.com/openlayers/openlayers/blob/v6.7.0/src/ol/control/FullScreen.js
+
+/**
+ * @return {boolean} Fullscreen is supported by the current platform.
+ */
+function isFullScreenSupported() {
+  if (!isInWindowScope) {
+    // Assume it is supported for most users.
+    return true;
+  }
+  const { body } = document;
+  return !!(
```

Comment:
Why a double negation?

Reply:
I copied this code from openlayers. It allows to cast to boolean: ` !!null === false`   and `!!{} === true`